src/test/java/net/snowflake/client/jdbc/StatementLatestIT.java

```diff
@@ -296,4 +297,42 @@ public void testQueryIdIsSetOnFailedExecuteQuery() throws SQLException {
       }
     }
   }
+
+  /**
+   * Test for setting query timeout on async queries. Applicable to versions after 3.20.0.
+   *
+   * @throws SQLException if there is an error when executing
+   */
+  @Test
+  public void testSetQueryTimeoutForAsnycQuery() throws SQLException {
+    try (Statement statement = connection.createStatement()) {
+      statement.unwrap(SnowflakeStatement.class).setParameter("MULTI_STATEMENT_COUNT", 0);
+      statement.setQueryTimeout(3);
```

Comment:
the connection is reused now in this test so when we set the timeout all the tests in this class will have the timeout - we should have local connection or reset query timeout in finally